lib/cstring.cpp

```diff
@@ -203,8 +203,7 @@ cstring cstring::before(const char *at) const { return substr(0, at - str); }
 
 cstring cstring::substr(size_t start, size_t length) const {
     if (size() <= start) return cstring::empty;
-    std::string s = str;
-    return s.substr(start, length);
+    return cstring(string_view().substr(start, length));
```

Comment:
Should this maybe return a `string_view` and have a constructor from a `string_view`?  Not sure if that would break anything.

Reply:
I was thinking about this. I'll try to take a look at that later, but I am not very happy about the idea of introducing a new implicit cast to cstring and without that I'm quite sure this would break something.